worker/crates/action-processor/src/plateau/attribute_flattener.rs

```diff
@@ -0,0 +1,553 @@
+                    let area_type_code_count = match area_type_code {
+                        Some(AttributeValue::Number(n)) => n.as_i64(),
+                        _ => None,
+                    };
+                    if desc.is_none()
+                        || area_type_code.is_none()
+                        || area_type_code_count.unwrap() > 2
+                    {
+                        continue;
+                    }
+
+                    let attrib = vec![
+                        (
+                            format!("土砂災害リスク_{}_区域区分", desc.unwrap()),
+                            area_type,
+                            "fme_buffer",
+                        ),
+                        (
+                            format!("土砂災害リスク_{}_区域区分コード", desc.unwrap()),
+                            area_type_code,
+                            "fme_uint16",
+                        ),
+                    ];
+
+                    for (k, v, t) in attrib {
+                        if let Some(value) = v {
+                            let attribute_name = Attribute::new(k.clone());
+                            feature.attributes.insert(attribute_name, value.clone());
+                            self.risk_to_attr_defs
+                                .entry("lsld".to_string())
+                                .or_default()
+                                .insert(k, t.to_string());
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+#[derive(Debug, Clone, Default)]
+pub struct AttributeFlattenerFactory;
+
+impl ProcessorFactory for AttributeFlattenerFactory {
+    fn name(&self) -> &str {
+        "PLATEAU.AttributeFlattener"
+    }
+
+    fn description(&self) -> &str {
+        "AttributeFlattener"
+    }
+
+    fn parameter_schema(&self) -> Option<schemars::schema::RootSchema> {
+        None
+    }
+
+    fn categories(&self) -> &[&'static str] {
+        &["PLATEAU"]
+    }
+
+    fn get_input_ports(&self) -> Vec<Port> {
+        vec![DEFAULT_PORT.clone()]
+    }
+
+    fn get_output_ports(&self) -> Vec<Port> {
+        vec![DEFAULT_PORT.clone()]
+    }
+
+    fn build(
+        &self,
+        _ctx: NodeContext,
+        _event_hub: EventHub,
+        _action: String,
+        _with: Option<HashMap<String, Value>>,
+    ) -> Result<Box<dyn Processor>, BoxedError> {
+        let process = AttributeFlattener {};
+        Ok(Box::new(process))
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct AttributeFlattener {}
+
+impl Processor for AttributeFlattener {
+    fn initialize(&mut self, _ctx: NodeContext) {}
+
+    fn num_threads(&self) -> usize {
+        5
+    }
+
+    fn process(
+        &mut self,
+        mut ctx: ExecutorContext,
+        fw: &mut dyn ProcessorChannelForwarder,
+    ) -> Result<(), BoxedError> {
+        let feature = &mut ctx.feature;
+
+        let mut flattener = Flattener::new();
```

Comment:
I think it would be better to make it so that it is new only once when it is built in the factory, since it is initialised every time a feature flows in.

Reply:
Moved it to AttributeFlattenerFactory.